sideboard/config.py

```diff
@@ -9,14 +10,72 @@
 from validate import Validator
 
 
+_defaults_re = re.compile(r'(.+)-defaults(\.\w+)$')
+
+
+def uniquify(x):
+    """
+    Returns an ordered copy of `x` with duplicate items removed.
+
+    >>> uniquify(['a', 'b', 'a', 'c', 'a', 'd', 'a', 'e'])
+    ['a', 'b', 'c', 'd', 'e']
+
+    """
+    seen = set()
+    return [i for i in x if i not in seen and not seen.add(i)]
```

Comment:
> On a separate note, what DOESN'T work is trying to uniquify a list that's out of order.

I think what Rob meant was that it would maintain the existing order rather than sort them.  I've requested a clarification in the docstring.

> I'm not sure I'm understanding this list comprehension, particularly the not seen.add(i).

The code is exactly equivalent to the following:

```python
xs = []
for i in x:
    if i not in seen:
        xs.append(i)
    seen.add(i)
return xs
```

The tricky part about implementing that as a list comprehension is the fact that we have to add the elements to ``seen`` as we go.  However, since ``seen.add(i)`` adds the current element to the set and always returns ``None`` no matter what, we can add it as a conditional in our list comprehension.